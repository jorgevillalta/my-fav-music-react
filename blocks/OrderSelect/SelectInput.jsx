import { FormControl, InputLabel, Select } from '@material-ui/core';
import * as React from 'react';
import styled from 'styled-components';

const SelectInput = styled(({ label, htmlFor, ...props }) => (
  <FormControl>
    <InputLabel htmlFor={htmlFor}>{label}</InputLabel>
    <Select {...props} />
  </FormControl>
))``;

export default SelectInput;
