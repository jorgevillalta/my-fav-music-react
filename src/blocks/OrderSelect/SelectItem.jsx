import MenuItem from '@material-ui/core/MenuItem';
import * as React from 'react';
import styled from 'styled-components';

const SelectItem = styled(({ ...props }) => <MenuItem {...props} />)``;

export default SelectItem;
