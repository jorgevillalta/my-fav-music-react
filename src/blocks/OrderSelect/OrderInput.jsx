import * as React from 'react';
import styled from 'styled-components';

import SelectInput from './SelectInput';

const OrderInput = styled(({ value, ...props }) => (
  <SelectInput
    label={'Order 🔝'}
    htmlFor={'order'}
    value={value || ''}
    inputProps={{
      name: 'order',
      id: 'select-order'
    }}
    {...props}
  />
))``;

export default OrderInput;
