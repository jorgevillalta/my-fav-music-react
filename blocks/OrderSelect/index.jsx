import styled from 'styled-components';

import OrderInput from './OrderInput';
import SelectInput from './SelectInput';
import SelectItem from './SelectItem';

const OrderSelect = styled.form`
  && {
    display: flex;
    align-content: stretch;
    align-items: center;

    /* childs margin */
    & > * {
      margin: 12px;
    }
  }
`;

OrderSelect.SelectInput = SelectInput;
OrderSelect.OrderInput = OrderInput;
OrderSelect.SelectItem = SelectItem;

export default OrderSelect;
