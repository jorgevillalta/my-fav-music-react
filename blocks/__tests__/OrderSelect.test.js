import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import * as React from 'react';

import OrderSelect from '../OrderSelect';

import 'jest-styled-components';

Enzyme.configure({ adapter: new Adapter() });

describe('OrderSelect Block', () => {
  let wrapper;
  const label = 'Order';
  const htmlFor = 'order';
  const props = { size: 'small', value: 'order' };

  test('Renders ok', () => {
    wrapper = shallow(<OrderSelect {...props} />);
    expect(wrapper.exists()).toBeTruthy();
  });

  test('Match snapshot', () => {
    wrapper = mount(<OrderSelect {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('SelectInput render props', () => {
    wrapper = mount(
      <OrderSelect.SelectInput label={label} htmlFor={htmlFor} {...props} />
    );
    expect(wrapper.prop('label')).toEqual(label);
    expect(wrapper.prop('htmlFor')).toEqual(htmlFor);
    expect(wrapper.prop('size')).toEqual(props.size);
  });

  test('OrderInput render props', () => {
    wrapper = mount(<OrderSelect.OrderInput {...props} />);
    expect(wrapper.prop('size')).toEqual(props.size);
  });

  test('SelectItem render props', () => {
    wrapper = mount(<OrderSelect.SelectItem {...props} />);
    expect(wrapper.prop('size')).toEqual(props.size);
  });
});
