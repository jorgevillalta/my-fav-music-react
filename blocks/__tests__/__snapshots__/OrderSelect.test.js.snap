// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSelect Block Match snapshot 1`] = `
.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.c0 > * {
  margin: 12px;
}

<styled.form
  size="small"
  value="order"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "OrderInput": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "SelectInput": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "SelectItem": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  && {
    display: flex;
    align-content: stretch;
    align-items: center;

    /* childs margin */
    & > * {
      margin: 12px;
    }
  }
",
          ],
        },
        "displayName": "styled.form",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "form",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    size="small"
    value="order"
  >
    <form
      className="c0"
      size="small"
      value="order"
    />
  </StyledComponent>
</styled.form>
`;
